homeassistant/components/loqed/config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Config flow for loqed integration."""
+from __future__ import annotations
+
+import json
+import logging
+from typing import Any
+
+import aiohttp
+from loqedAPI import loqed
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import webhook
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.const import CONF_HOST, CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(
+    hass: HomeAssistant, data: dict[str, Any], zeroconf_host: str | None
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    newdata = data
+    json_config = json.loads(data["config"])
+    newdata["ip"] = json_config["bridge_ip"]
+    newdata["host"] = json_config["bridge_mdns_hostname"]
+    newdata["bkey"] = json_config["bridge_key"]
+    newdata["key_id"] = int(json_config["lock_key_local_id"])
+    newdata["api_key"] = json_config["lock_key_key"]
+
+    if zeroconf_host is not None and zeroconf_host != newdata["host"]:
+        raise InvalidAuth(
+            f"Got config for {newdata['host']} while configuring {zeroconf_host} "
+        )
+
+    # 1. Checking loqed-connection
+    try:
+        session = async_get_clientsession(hass)
+
+        apiclient = loqed.APIClient(session, "http://" + newdata["ip"])
+        api = loqed.LoqedAPI(apiclient)
+        lock = await api.async_get_lock(
+            newdata["api_key"], newdata["bkey"], newdata["key_id"], newdata["host"]
+        )
+        newdata["id"] = lock.id
+        # checking getWebooks to check the bridgeKey
+        await lock.getWebhooks()
+    except (aiohttp.ClientError):
+        _LOGGER.error("HTTP Connection error to loqed lock")
+        raise CannotConnect from aiohttp.ClientError
+    return newdata
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for loqed."""
+
+    VERSION = 2
+
+    def __init__(self) -> None:
+        """Initialize the ConfigFlow for the LOQED integration."""
+        self._host: str | None = None
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        self._host = discovery_info.hostname.rstrip(".")
+
+        session = async_get_clientsession(self.hass)
+        apiclient = loqed.APIClient(session, f"http://{self._host}")
+        api = loqed.LoqedAPI(apiclient)
+        lock_data = await api.async_get_lock_details()
+
+        # Check if already exists
+        await self.async_set_unique_id(lock_data["bridge_mac_wifi"])
+        self._abort_if_unique_id_configured()
```

Comment:
Not sure if the lock can change hostname. It's a unique Id for as far as I can tell

Reply:
The hostname would not be guaranteed. mDNS collisions, router changes, IP addresses instead of names. And all of them can change regardless of the device and software of the lock.

The `updates` parameter allows updating the host whenever it occurs. The general best practice is to do that.

Reply:
Added updates parameter